src/NServiceBus.Core/Transports/Msmq/MsmqUtilities.cs

```diff
@@ -15,6 +17,29 @@ namespace NServiceBus
 
     class MsmqUtilities
     {
+        static MsmqUtilities()
+        {
+            var getRawBytesSegment = new DynamicMethod(nameof(GetAsArraySegment), typeof(ArraySegment<byte>), new[]{typeof(MemoryStream)}, true);
```

Comment:
If I recall correctly this would not work on .NET Core. But OK since MSMQ remains a full framework thing anyway.